### DISCUSSION TITLE
Consumer Batch Processing

### DISCUSSION THREAD

**Discussion by udiazulayintel** [rabbitmq/rabbitmq-server#9119] [Other]
Hi,

I have created a new consumer over a queue with prefetch batch of 100 messages, in order to send a bulk insert to SQL Server
and when the publisher published 199 messages, I see 99 messages stuck and are unacknowledged waiting for the 100 batch in order to be bulk insert into SQL server.

Please advise how can I resolve it ?

**Comment by kjnilsson** [comment]
If you only insert every 100 messages you will keep the 99 in a prefetch
buffer assigned to the client until either the client disconnects or you
receive 1 more message.

I suggest your client also impl a timeout where it inserts the messages it
has got even if it is fewer than 100.

Cheers
Karl

On Wed, 16 Aug 2023 at 12:52, Udi Azulay ***@***.***> wrote:

> Hi,
>
> I have created a new consumer over a queue with prefetch batch of 100
> messages, in order to send a bulk insert to SQL Server
> and when the publisher published 199 messages, I see 99 messages stuck and
> are unacknowledged waiting for the 100 batch in order to be bulk insert
> into SQL server.
>
> Please advise how can I resolve it ?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-server/discussions/9119>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJAHFBA5YWRPFZXS75VXSLXVSYAZANCNFSM6AAAAAA3SNEHRA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
*Karl Nilsson*

**Comment by udiazulayintel** [comment]
Thanks,
Indeed this is what i did, is this the best practice ? to have a timer over my consumer Task ?
